### PR TITLE
Use ScrollBars from QQC2, which supports mouse and touch events

### PR DESCRIPTION
**Pull request by vladest**
Test plan:
1. Open Tablet
2. Open Running scripts
3. Make sure scroll bar works using mouse/stylus/lasers
4. Open any File dialog, make sure its possible scroll lists using scrollbars
5. Click on empty spaces on scroll bars, make sure lists gets scrolled
6. Repeat same for Desktop mode

Known issues:
leaving Tablet with pressed button and release outside Tablet will cause stick to mouse cursor. Its common problem not only for scroll bars but for whole Tablet area (just try to click and scroll and list moving mouse outside Tablet)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5105/HighFidelity-Beta-PR11858-0ea5a7635ce3896c84752527789df7b3515bb651.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5105/HighFidelity-Beta-PR11858-0ea5a7635ce3896c84752527789df7b3515bb651.dmg

**Comment by themelissabrown** [comment]
build this please

**Comment by YuppyPlays** [comment]
Awaiting for newer build for QA.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5106/HighFidelity-Beta-PR11858-0ea5a7635ce3896c84752527789df7b3515bb651.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5106/HighFidelity-Beta-PR11858-0ea5a7635ce3896c84752527789df7b3515bb651.dmg

**Comment by YuppyPlays** [comment]
Testing.

**Comment by themelissabrown** [comment]
Will assess this PR once https://github.com/highfidelity/hifi/pull/11834 has been merged

**Comment by YuppyPlays** [comment]
Testing complete:

Open Tablet **Check**
Open Running scripts **Check**
Make sure scroll bar works using mouse/stylus **Check**
Open any File dialog, make sure scroll bars still works **Fail, could not find a path long enough as I could not click on any folders/items with stylus or mouse- however, upon a restart I can confirm I was able to scroll and click items normally, though I noticed particularly with the Stylus- it would scroll before I even touched the scroll bar (like it detected it even at 2-3 inches away from scroll)**

**Comment by ZappoMan** [comment]
1 - Unix build is failing because of `variable ‘button’ set but not used`
2 - This PR also needs to be tested against stylus + laser, and we need to confirm that this doesn't introduce performance problems. This needs a better test plan.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5110/HighFidelity-Beta-PR11858-a56e4e10c30d6bba832cd1b2f46b834ff8fd91fe.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5110/HighFidelity-Beta-PR11858-a56e4e10c30d6bba832cd1b2f46b834ff8fd91fe.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5147/HighFidelity-Beta-PR11858-73a557a8a4d8a4a4a4905c91fe3f5f64c91a6e5b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5152/HighFidelity-Beta-PR11858-6136a4cea6779189c2effeb0da81422a6a7dcac7.dmg

**Comment by themelissabrown** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5177/HighFidelity-Beta-PR11858-6136a4cea6779189c2effeb0da81422a6a7dcac7.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5177/HighFidelity-Beta-PR11858-6136a4cea6779189c2effeb0da81422a6a7dcac7.dmg

**Comment by YuppyPlays** [comment]
@vladest is this due a re-test? with currently created build? :)

File: interface/resources/qml/controls-uit/Table.qml [hunk 1]
```diff
@@ -24,6 +25,54 @@ TableView {
...
+        anchors.bottom: tableView.bottom
+
+        background: Item {
+            implicitWidth: hifi.dimensions.scrollbarBackgroundWidth
+            Rectangle {
+                anchors {
+                    fill: parent;
+                    topMargin: tableView.headerVisible ? hifi.dimensions.tableHeaderHeight : 0
+                }
+                radius: hifi.dimensions.scrollbarHandleWidth/2
```
Comment (SamGondelman) [inline interface/resources/qml/controls-uit/Table.qml]: I don't think the background is supposed to be rounded.  Or maybe this is why we were extending the top and bottom before.  If you select the top item in a list now, you can see behind the scrollbar.

**Comment by themelissabrown** [comment]
@ZappoMan @conklin94122 is there a testrails plan that would work well to be a comprehensive test for this?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5214/HighFidelity-Beta-PR11858-94937374c5d25d1d22804be801d78b53eaaa9bf2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5214/HighFidelity-Beta-PR11858-94937374c5d25d1d22804be801d78b53eaaa9bf2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5219/HighFidelity-Beta-PR11858-7283f0127b94b64348565181baafa6d20b141706.exe

**Comment by themelissabrown** [comment]
New test plan: https://highfidelity.testrail.net/index.php?/suites/view/27

**Comment by conklin94122** [comment]
https://highfidelity.testrail.net/index.php?/runs/view/1229

**Comment by Nex-Pro** [comment]
Testing Win10

**Comment by Nex-Pro** [comment]
### FAIL

https://highfidelity.testrail.net/index.php?/runs/view/1229&group_by=cases:section_id&group_order=asc

**Comment by Nex-Pro** [comment]
In addition on every occasion that I changed from desktop to HMD I had to reset sensors as my controllers were behind and to one side of the avatar and view was in opposite direction to avatar model.

**Comment by SamGondelman** [comment]
https://highfidelity.manuscript.com/f/cases/6625/Asset-browser-vertical-scrollbar-not-working-in-tablet-for-desktop-and-HMD
https://highfidelity.manuscript.com/f/cases/8545/HMD-Unable-to-Scroll-Script-selection-UI
https://highfidelity.manuscript.com/f/cases/9726/HMD-Asset-Scroll-bar-is-not-functional-when-using-hand-controllers

**Comment by Nex-Pro** [comment]
Testing this by request when ready.

**Comment by Nex-Pro** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5276/HighFidelity-Beta-PR11858-a3d406d8d1718a9b2fa44eaf193f616c973791ef.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5276/HighFidelity-Beta-PR11858-a3d406d8d1718a9b2fa44eaf193f616c973791ef.dmg

**Comment by Nex-Pro** [comment]
Testing

**Comment by Nex-Pro** [comment]
### PASS

Test plan:

Open Tablet

Open Running scripts

Make sure scroll bar works using mouse/stylus/lasers **SUCCESS**

Open any File dialog, make sure its possible scroll lists using scrollbars **SUCCESS**

Click on empty spaces on scroll bars, make sure lists gets scrolled **SUCCESS**

Repeat same for Desktop mode **SUCCESS**

Known issues:

leaving Tablet with pressed button and release outside Tablet will cause stick to mouse cursor. 
It's a 	common problem not only for scroll bars but for whole Tablet area (just try to click and scroll and list moving mouse outside Tablet)

Good work @vladest  We needed this fix, thanks.

**Pull request merged**